include/deal.II/base/function_parser.h

```diff
@@ -26,10 +26,7 @@
 #include <vector>
 #include <map>
 
-namespace mu
-{
-  class Parser;
-}
+#include <muParser.h>
```

Comment:
I don't really see how MSVC does something wrong here. When we instantiate that template it creates functions like `mu::Parser & std::vector<mu::Parser>::back()`; this requires a declaration of the value type to be available. I suppose GCC manages to get around this somehow.